### PR TITLE
op.sh: lint

### PR DESCRIPTION
**Pull request by maxime-desroches**

### CODE REVIEW

**Comment by sshane** [comment]
I added this to the end of card.py, and mypy passed:

```
def test(a: int):
  print(a + '2')
```

Running mypy on this file took at least 5 seconds alone, so perhaps it's not being detected as a changed file.

Edit: I did `git add card.py` and `op lint` detected the change now. I feel pre-commit would have checked the file in this case as it's already checked into the repo?

**Comment by maxime-desroches** [comment]
@sshane If `card.py` is already in the repo, running `pre-commit run --all` will flag the error. However, the pre-commit running during `git commit` will not flag the error (unless you staged your changes to `card.py`).

I guess you are right that we should support both options especially if we want to put the new `lint.sh` in a pre-commit hook

File: .github/workflows/selfdrive_tests.yaml [hunk 1]
```diff
@@ -61,13 +61,20 @@ jobs:
         cd $STRIPPED_DIR
         ${{ env.RUN }} "release/check-dirty.sh && \
                         MAX_EXAMPLES=5 $PYTEST -m 'not slow' selfdrive/car"
-    - name: static analysis
+    - name: Getting modified files
```
Comment (adeebshihadeh) [inline .github/workflows/selfdrive_tests.yaml:64]: CI should always run on all files, like pre-commit used to do

**Review by adeebshihadeh — COMMENTED** [review]
is mypy the only thing that's slow enough to require only running on the diff?

**Comment by maxime-desroches** [comment]
For the current tests:
Mypy takes ~23 seconds for the whole tree
Codespell takes ~1.5 seconds for the whole tree
The rest is instantaneous

File: .github/workflows/selfdrive_tests.yaml (hunk 1 above)
Comment (maxime-desroches) [reply .github/workflows/selfdrive_tests.yaml:64]: Wouldn't that break the new goal of having the same pre-commit result in CI and locally?

**Comment by sshane** [comment]
It's just a little weird that you need to stage your changes to a file for it to run, but then any future changes are technically unstaged, but it will still run on that file. I think pre-commit would run on all tracked files regardless of unstaged changes.

**Comment by maxime-desroches** [comment]
`op lint` now runs on all files staged or committed in the tree (~20s without mypy cache)
`op lint --fast` to run only fast checks (~0.35s)
`op lint --skip mypy` to skip a test

**Pull request merged**